crates/openvino/src/blob.rs

```diff
@@ -124,6 +124,28 @@ impl Blob {
         Ok(slice)
     }
 
+    /// Retrieve the [`Blob`]'s data as an immutable slice of type `T`.
+    ///
+    /// # Safety
+    ///
+    /// This function is `unsafe`, since the values of `T` may not have been properly initialized;
+    /// however, this functionality is provided as an equivalent of what C/C++ users of OpenVINO
+    /// currently do to access [`Blob`]s with, e.g., floating point values:
+    /// `results.buffer_as_type::<f32>()`.
+    pub unsafe fn buffer_as_type<T>(&self) -> Result<&[T]> {
+        let mut buffer = Blob::empty_buffer();
+        InferenceError::from(ie_blob_get_buffer(
+            self.instance,
+            std::ptr::addr_of_mut!(buffer),
+        ))?;
+        // This is very unsafe, but very convenient: by allowing users to specify T, they can
+        // retrieve the buffer in whatever shape they prefer. But we must ensure that they cannot
+        // read too many bytes, so we manually calculate the resulting slice `size`.
+        let size = self.byte_len()? / std::mem::size_of::<T>();
+        let slice = std::slice::from_raw_parts(buffer.__bindgen_anon_1.buffer.cast::<T>(), size);
+        Ok(slice)
```

Comment:
It would be ideal to de-duplicate this but the `from_raw_parts` is different so don't worry about it.